Added interactive navigations to quickly find failed tests and reasons

judovana opened the pull request:
- possibility to hide/show
   - passed/failed/ignored/bailed out tests
   - diagnostic details
   - individual diagnostic items
   -  the three state buttons can control individual items
   - several preset views to mimic the major workflows
- number is now direct url to itself

precautions against CVE 3190 should be applied
tested in firefox and chromium


rebased https://github.com/jenkinsci/tap-plugin/pull/29

judovana commented:
Same failure in CI.  It seems https://github.com/judovana/tap-plugin/blob/bddc3aa78250ec84bedd1204a1ea54a17e516511/src/main/webapp/interactive.js#L311 will need some love.  Will elaborate (which do not prevent anyone to start reviewing O:)
j.

judovana commented:
> Same failure in CI. It seems https://github.com/judovana/tap-plugin/blob/bddc3aa78250ec84bedd1204a1ea54a17e516511/src/main/webapp/interactive.js#L311 will need some love. Will elaborate (which do not prevent anyone to start reviewing O:) j.

Actually, `function getSelectState(...clazz) {`  It seems that the in-test-browser can not read this JS syntax:(

judovana commented:
Ok, seems the fix was indeed in signature of that method. Test passed for me, and all seems working. But will double check at the morning. Happy hacking!

judovana commented:
Hello! There is a month anniversary.  Will you be willing to take a look please?

judovana commented:
Ping please?

judovana commented:
@kinow  You there please?

judovana commented:
@kinow - As understood in https://issues.jenkins.io/browse/JENKINS-73484 the unittests were blocker of thiis PR. Thus I started adding them. Please, if anyhting is blocking so long, but still live PR, jsut tell WHAT.  Silecne wold not help. 

I had added basic tests for checking some static changes in code. Will add few more, and then also soem of the JS interactions if I figure how. Please step into rather sooner then later.

judovana commented:
@kinow in this way I can continue to add ny tests you need, but that do not seem exactly right until we have agreement on impl.

The ball is now in your hand.

smlambert commented:
I am interested in this enhancement to the tap-plugin, and wondering if there is an ETA for it, or any other concerns before bringing it into the code base?

judovana committed:
Added interactive navigations to quickly find failed tests and reasons

- possibility to hide/show
   - passed/failed/ignored/bailed out tests
   - diagnosticd details
   - individual diagnostic items
   -  the three state buttons can control individual items
   - several preset views to mimic the major workflows
- number is now direct url to itself

precautions against CVE 3190 should be applied
tested in firefox and chromium

judovana commented:
@kinow - san!  It is yet again rebased. As I'm moving again to teams which are using tap plugin heavily, and can nto run from random forks, can you please look into that?
